torch/onnx/_internal/io_adapter.py

```diff
@@ -487,3 +540,44 @@ def apply(self, model_outputs: Any) -> Sequence[Any]:
                 error_message="Model outputs incompatible with the format that was exported. ",
             )
         return flattened_outputs
+
+
+class PrependParamsAndBuffersAotAutogradInputStep:
+    """Prepend model parameters and buffers to the user input.
+
+    :func:`torch.export.export` lifts model parameters and buffers as model input, thus, they
+    must be added to the user input before the model  is executed.
```

Comment:
```suggestion
    must be added to the user input before the model is executed.
```
nit

Reply:
Are the prepended params and buffers part of onnx model inputs? Do people need to supply them through input_dict for ort.run call?

Reply:
Yes, they are. They are supplied through `ort.run`

As an example, 

```python
        class Model(torch.nn.Module):
            def __init__(self):
                self.normal = torch.distributions.normal.Normal(0, 1)
                super().__init__()

            def forward(self, x):
                return self.normal.sample(x.shape)

        x = torch.randn(2, 3)
        exported_program = torch.export.export(Model(), args=(x,))
        _ = torch.onnx.dynamo_export(
            exported_program,
            x,
        )
```
results in a `GraphModule`

```python
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, arg0_1: f32[], arg1_1: f32[], arg2_1: f32[2, 3]):
            # File: /home/justinchu/anaconda3/envs/onnx/lib/python3.10/site-packages/torch/distributions/normal.py:70, code: return torch.normal(self.loc.expand(shape), self.scale.expand(shape))
            expand: f32[2, 3] = torch.ops.aten.expand.default(arg0_1, [2, 3]);  arg0_1 = None
            expand_1: f32[2, 3] = torch.ops.aten.expand.default(arg1_1, [2, 3]);  arg1_1 = None
            normal: f32[2, 3] = torch.ops.aten.normal.Tensor_Tensor(expand, expand_1);  expand = expand_1 = None
            return (normal,)

Graph signature: ExportGraphSignature(parameters=[], buffers=['L__self___normal_loc', 'L__self___normal_scale'], user_inputs=['arg2_1'], user_outputs=['normal'], inputs_to_parameters={}, inputs_to_buffers={'arg0_1': 'L__self___normal_loc', 'arg1_1': 'L__self___normal_scale'}, buffers_to_mutate={}, backward_signature=None, assertion_dep_token=None)
Range constraints: {}
Equality constraints: []
```

with 1 user input and 2 buffers. When converting that as a ONN graph, we have all 3 as inputs:

```bash
[name: "arg0_1"
type {
  tensor_type {
    elem_type: 1
    shape {
    }
  }
}
, name: "arg1_1"
type {
  tensor_type {
    elem_type: 1
    shape {
    }
  }
}
, name: "arg2_1"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_value: 2
      }
      dim {
        dim_value: 3
      }
    }
  }
}
]
```

Reply:
I think we should put those back as initializers, can be in follow up.

Reply:
That is a good point. Will do it separately. 

Adding it back makes the ONNX file complete and executable by any runtime, but then we have the issue with models >2GB. On the other hand, if the user wants to save the onnx model (and discard `ExportOutput`), then all saved buffers/initializers are lost. Maybe we have to serialize `ExportOutput` in the  ExportedProgram` as input approach